stock_internal_transfer/stock_internal_transfer.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp.osv import orm, fields
+from openerp.netsvc import LocalService
+import openerp.addons.decimal_precision as dp
+from openerp.tools.translate import _
+
+
+class StockInternalTransferLine(orm.TransientModel):
+    _name = "stock.internal.transfer.line"
+    _rec_name = 'product_id'
+    _columns = {
+        'product_id': fields.many2one('product.product', string="Product",
+                                      required=True, ondelete='CASCADE'),
+        'quantity': fields.float(
+            "Quantity",
+            digits_compute=dp.get_precision('Product Unit of Measure'),
+            required=True),
+        'product_uom': fields.many2one('product.uom', 'Unit of Measure',
+                                       required=True, ondelete='CASCADE'),
+        'wizard_id': fields.many2one('stock.internal.transfer',
+                                     string="Wizard", ondelete='CASCADE'),
+    }
+
+    def onchange_product_id(self, cr, uid, ids, product_id, context=None):
+        uom_id = False
+        if product_id:
+            product = self.pool['product.product'].\
+                browse(cr, uid, product_id, context=context)
+            uom_id = product.uom_id.id
+        return {'value': {'product_uom': uom_id}}
+
+
+class StockInternalTransfer(orm.TransientModel):
+    _name = 'stock.internal.transfer'
+
+    def _get_transfer_mode_sel(self, cr, uid, context=None):
+        return [('default', _('Default')), ('force', _('Force Transfer'))]
+
+    _columns = {
+        'location_id': fields.many2one(
+            'stock.location', string="Location", required=True,
+            ondelete='CASCADE', domain=[('usage', '<>', 'view')]),
+        'location_dest_id': fields.many2one(
+            'stock.location', string="Dest. Location", required=True,
+            ondelete='CASCADE', domain=[('usage', '<>', 'view')]),
+        'delivery_partner_id': fields.many2one(
+            'res.partner', string="Delivery Address", required=True),
```

Comment:
The initial idea here was to have the final destination on all delivery slips. E.g. in the transit version or other inherited versions of this module the intermediate locations could have completly different address or even no address being a truck or ship. But you are right it should get set internally automatically and shouldn't be required. It shouldn't be in the wizard at all I think. 
I will remove this from the wizard and get the partner from the destination location.